rest-server/engines/src/main/scala/com/actionml/engines/urnavhinting/URNavHintingAlgorithm.scala

```diff
@@ -268,10 +268,10 @@ class URNavHintingAlgorithm private (engine: URNavHintingEngine, initParams: Str
       "spark.mongodb.input.collection" -> dataset.getIndicatorEventsCollectionName
     )
 
-    SparkContextSupport.getSparkContext(initParams, defaults).map { implicit sc =>
+    SparkContextSupport.getSparkContext(initParams, defaults, Array(classOf[URNavHintingEvent])).map { implicit sc =>
```

Comment:
What uses the Array of classes? Kryo? If so what classes need to be passed? This needs to be documented somewhere.

Reply:
for instance what if we had to have 2 readRdds from 2 DAO collections? This is the case for the full UR that we do not need for URNavHinting.

What other times do we need to include classes in this array? It is called `kryoClasses`. Ok but under what circumstances do we need to pass them in rather than simply inherit `Serializable`?

Reply:
kryo types should be set explicitly as a spark context setting. In case of URNavHintingAlgorithm there is a computation on cluster that requires serialization/deserialization of URNavHintingEvent. That's why URNavHintingEvent must be registered on spark context that way.